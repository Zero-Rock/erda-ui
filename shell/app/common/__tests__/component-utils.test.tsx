import React from 'react';
import { shallow } from 'enzyme';
import { getLabel } from '../utils/component-utils';
import { describe, it } from '@jest/globals';

describe('getLabel', () => {
  const label = 'i am a label';
  const tips = 'i an a tips';
  it('should only label', () => {
    expect(getLabel(label)).toBe(label);
  });
  it('should not show tips', () => {
    const Comp = getLabel(label, tips, false);
    const wrapper = shallow(<div>{Comp}</div>);
    expect(wrapper.find('span').text()).toContain(label);
    expect(wrapper).not.toContain(label);
  });
  it('should not show tips', () => {
    const Comp = getLabel(label, tips, true);
    const wrapper = shallow(<div>{Comp}</div>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('span').at(0).text()).toContain(label);
    expect(wrapper.find('span').at(1).text()).toContain('*');
    expect(wrapper.find('Tooltip').prop('title')).toBe(tips);
  });
});
