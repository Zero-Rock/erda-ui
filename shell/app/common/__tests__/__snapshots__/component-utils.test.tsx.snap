// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLabel should not show tips 1`] = `
<div>
  <span>
    <span
      style={
        Object {
          "color": "red",
          "marginRight": 4,
        }
      }
    >
      *
    </span>
    i am a label
    Â 
    <Tooltip
      allow={false}
      arrow={false}
      arrowPointAtCenter={false}
      autoAdjustOverflow={true}
      mouseEnterDelay={0.1}
      mouseLeaveDelay={0.1}
      placement="top"
      prefixCls="pk-tooltip"
      title="i an a tips"
      transitionName="zoom-big-fast"
      type="shallow"
    >
      <Icon
        className="color-text-icon"
        type="question-circle-o"
      />
    </Tooltip>
  </span>
</div>
`;
